### PR TITLE
add CI workflows to repository

### PR DESCRIPTION
**Pull request by Salka1988**
closes https://github.com/FuelLabs/fuels-types/issues/2

~blocked by: https://github.com/FuelLabs/fuels-types/issues/1~

### CODE REVIEW

**Review by segfault-magnet — COMMENTED** [review]
This PR seems to replicate the work done in https://github.com/FuelLabs/fuels-types/pull/3. 

Why not close it and make this PR close two issues?

**Comment by Salka1988** [comment]
> This PR seems to replicate the work done in #3.
> 
> Why not close it and make this PR close two issues?

There's no need. We will go step by step as defined in https://github.com/fuelabs/fuels-rs/isueš/752. CI is added in this step.

**Comment by mohammadfawaz** [comment]
You can create the branch for this PR _from_ the branch of the other PR and then re-create this PR with only the differential between the two branches (instead of comparing this with `master`). Then turn this PR into a draft. Once the base PR is reviewed and merged, this PR will be automatically updated to compare with `master` and then you can mark it as ready for review.

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: github pages
```
Comment (digorithm) [inline .github/workflows/gh-pages.yml:1]: I don't think we need this action at all because we're not publishing live docs for this project.

**Review by digorithm — CHANGES_REQUESTED** [review]

File: .github/workflows/gh-pages.yml (hunk 1 above)
Comment (Salka1988) [reply .github/workflows/gh-pages.yml]: Deleted!

**Review by digorithm — APPROVED** [review]
:shipit:

**Pull request merged**